### PR TITLE
permissions-center: add basic permission sync jobs API.

### PR DESCRIPTION
**Pull request by sashaostrikov**
Test plan:
Tests added.

Closes https://github.com/sourcegraph/sourcegraph/issues/47301

### CODE REVIEW

File: cmd/frontend/graphqlbackend/permission_sync_jobs.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	NumFailures() int
+	LastHeartbeatAt() time.Time
+	ExecutionLogs() []executor.ExecutionLogEntry
+	WorkerHostname() string
+	Cancel() bool
+	RepositoryID() graphql.ID
+	UserID() graphql.ID
+	Priority() database.PermissionSyncJobPriority
+	NoPerms() bool
+	InvalidateCaches() bool
```
Comment (thenamankumar) [inline cmd/frontend/graphqlbackend/permission_sync_jobs.go:40]: Add permissionsAdded, Removed, Found.

Comment (sashaostrikov) [reply cmd/frontend/graphqlbackend/permission_sync_jobs.go:40]: oh, they're merged, right! 👍🏻

File: enterprise/cmd/frontend/internal/authz/resolvers/permission_sync_jobs.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+func (p *permissionSyncJobResolver) State() string {
+	return p.job.State
+}
+
+func (p *permissionSyncJobResolver) FailureMessage() *string {
+	return p.job.FailureMessage
+}
+
+func (p *permissionSyncJobResolver) Reason() database.PermissionSyncJobReason {
+	return p.job.Reason
```
Comment (thenamankumar) [inline enterprise/cmd/frontend/internal/authz/resolvers/permission_sync_jobs.go:100]: TODO: in upcoming PR, it should be converted into a ReasonType. 

{ group, message }

**Review by thenamankumar — APPROVED** [review]

File: cmd/frontend/graphqlbackend/permission_sync_jobs.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	ID() graphql.ID
+	State() string
+	FailureMessage() *string
+	Reason() database.PermissionSyncJobReason
+	CancellationReason() string
+	TriggeredByUserID() int32
+	QueuedAt() time.Time
+	StartedAt() time.Time
+	FinishedAt() time.Time
+	ProcessAfter() time.Time
```
Comment (mrnugget) [inline cmd/frontend/graphqlbackend/permission_sync_jobs.go:29]: Shouldn't these be pointers so they can be nullable in the API?

Comment (mrnugget) [reply cmd/frontend/graphqlbackend/permission_sync_jobs.go:29]: Side-note: I personally find it really hard to come up with the correct resolvers without having the GraphQL schema first. So I'd suggest always starting with that

**Pull request merged**